Source/Microsoft.Teams.Apps.FAQPlusPlus.Common/Providers/QnaServiceProvider.cs

```diff
@@ -183,19 +183,31 @@ public async Task DeleteQnaAsync(int questionId)
         /// </summary>
         /// <param name="question">Question text.</param>
         /// <param name="isTestKnowledgeBase">Prod or test.</param>
-        /// <returns>QnaSearchResult result as response.</returns>
-        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase)
+        /// <param name="isPrompt">Is the question from a follow up prompt.</param>
+        /// <param name="previousQnAId">Id of previous question.</param>
+        /// <param name="previousUserQuery">Previous question information.</param>
+        /// <returns>QnaSearchResultList result as response.</returns>
+        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase, bool isPrompt = false, string previousQnAId = null, string previousUserQuery = null)
```

Comment:
isPrompt seems redundant here-- isn't it the case that isPrompt is true if and only if previousQnAId is not null (since it's a follow-up)?

Reply:
Removed the parameter and the check. Previously added for extra validation.